### ISSUE TITLE
Empty response from server

### ISSUE THREAD

**Issue by B4yM4x** [hashicorp/docker-vault#28]
Hello,
I am trying to use vault in docker but I am not able to authenticate via APIs with any users I create. I get an "empty response from server".

This is the sequence of commands I run

`docker run -p 8200:8200 --cap-add=IPC_LOCK -e 'VAULT_DEV_ROOT_TOKEN_ID=myroot' -e 'VAULT_DEV_LISTEN_ADDRESS=127.0.0.1:8200’ vault
docker exec -it DOCKER_VAULT_ID vault status -address=http://127.0.0.1:8200
3. docker exec -it DOCKER_VAULT_ID vault auth -address=http://127.0.0.1:8200   
4. docker exec -it DOCKER_VAULT_ID vault auth-enable -address=http://127.0.0.1:8200 userpass
5. docker exec -it DOCKER_VAULT_ID vault write -address=http://127.0.0.1:8200 auth/userpass/users/new_user password=new_password policies=admins
`

If I try to authenticate locally via the CLI for new_user it works, but with the REST APIs I get an empty response.

Please note that If I run however vault locally, on my machine, and I repeat exactly all the steps above  (skipping point 3) I get good results.

Any suggestion?
Thanks!

**Comment by gnilchee** [comment]
@B4yM4x Are you running the commands from from within the Docker container or from your computer? 

I suspect the 127.0.0.1 address you are declaring is binding it to the loopback interface and the kernel is probably reseting your connection from outside of the container. 

Since this was posted some time ago if you are having issues take a look at my repo: https://github.com/gnilchee/vault. 

The setup works consistently for me and it is based off the 0.7.0 Vault container. I have followed my README multiple times and updated it as I found issues. I hope this helps!
